### ISSUE TITLE
Uninstalling windows hosting bundle not working

### ISSUE THREAD

**Issue by dfischerinsiderstechnologies** [dotnet/aspnetcore#50826]
### Description

I have 3 Versions of windows hosting bundle installed on Windows. Running "dotnet-core-uninstall list" gave me this:

`                                                         
This tool cannot uninstall versions of the runtime or SDK that are 
    - SDKs installed using Visual Studio 2019 Update 3 or later.
    - SDKs and runtimes installed via zip/scripts.
    - Runtimes installed with SDKs (these should be removed by removing that SDK).
The versions that can be uninstalled with this tool are:

.NET Core SDKs:
  6.0.414  x64    [Used by Visual Studio. Specify individually or use --force to remove]
  6.0.122  x64
  2.1.526  x64    [Used by Visual Studio 2017. Specify individually or use --force to remove]

.NET Core Runtimes:
  6.0.22  x86
  6.0.22  x64
  6.0.16  x64

ASP.NET Core Runtimes:
  6.0.22  x86
  6.0.22  x64
  5.0.16  x86
  5.0.16  x64

.NET Core Runtime & Hosting Bundles:
  6.0.22
  6.0.21
  6.0.16`

I want to uninstall all of the listed versions "6.0.22", "6.0.21" and "6.0.16"
running "dotnet-core-uninstall remove --hosting-bundle 6.0.22 --hosting-bundle 6.0.21 --hosting-bundle 6.0.16" Gave me:

`The following items will be removed:
  Microsoft .NET 6.0.22 - Windows Server Hosting
  Microsoft .NET 6.0.21 - Windows Server Hosting
  Microsoft .NET 6.0.16 - Windows Server Hosting

To avoid breaking Visual Studio or other problems, read https://aka.ms/dotnet-core-uninstall-docs.

Do you want to continue? [y/n] y
Uninstalling: Microsoft .NET 6.0.22 - Windows Server Hosting.
Uninstalling: Microsoft .NET 6.0.21 - Windows Server Hosting.
Uninstalling: Microsoft .NET 6.0.16 - Windows Server Hosting.
The uninstallation operation failed. Exit code was 1 for ""C:\ProgramData\Package Cache\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}\dotnet-hosting-6.0.16-win.exe" /uninstall /quiet".`

Unfortunately all 3 Versions are still installed after this command, even though only the installation for version 6.0.16 seemed to fail.

I then tried running ".\dotnet-hosting-6.0.16-win.exe /uninstall" in the "C:\ProgramData\Package Cache\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}"-Folder, which gave me more information:

`[2E70:28A0][2023-09-15T13:40:57]i001: Burn v3.14.0.5722, Windows v10.0 (Build 19045: Service Pack 0), path: C:\ProgramData\Package Cache\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}\dotnet-hosting-6.0.16-win.exe
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'BundleNameShort' to value 'Microsoft .NET 6.0.16'
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'BundleNameSub' to value 'Windows Server Hosting'
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'InstallResetIISState' to value 'disable'
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'InstallNoIISState' to value 'disable'
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'ModifyResetIISState' to value 'disable'
[2E70:28A0][2023-09-15T13:40:57]i000: Initializing string variable 'ModifyNoIISState' to value 'disable'
[2E70:28A0][2023-09-15T13:40:57]i009: Command Line: '"-burn.clean.room=C:\ProgramData\Package Cache\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}\dotnet-hosting-6.0.16-win.exe" -burn.filehandle.attached=548 -burn.filehandle.self=560 /uninstall'
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\DFi\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20230915134057.log'
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'WixBundleManufacturer' to value 'Microsoft Corporation'
[2E70:4110][2023-09-15T13:40:57]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[2E70:4110][2023-09-15T13:40:57]i000: Setting version variable 'WixBundleFileVersion' to value '6.0.16.23174'
[2E70:28A0][2023-09-15T13:40:57]i100: Detect begin, 7 packages
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'VersionNT64' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Product or related product not found: {80AAA3ED-9565-4858-A58A-9AD5C50BD8E3}
[2E70:28A0][2023-09-15T13:40:57]i000: Setting version variable 'DotNetRedistLtsProductVersion_x64' to value '0.0.0.0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT VersionNT64' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'IISCoreWebEngineInstalled_x86' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'IISW3SVCInstalled_x86' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'IISCoreWebEngineInstalled_x64' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'IISW3SVCInstalled_x64' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1)))' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i052: Condition '((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1))' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'InstallResetIISState' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1)))' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i052: Condition '((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1))' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'ModifyResetIISState' to value '1'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_ANCM_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_FTS_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_LTS_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_RUNTIME_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_SHARED_CONFIG_CHECK_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_SHAREDFX_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting numeric variable 'OPT_NO_X86_Exists' to value 1
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'VersionNT64' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting version variable 'SharedFxRedistProductVersion_x64' to value '6.0.16.23174'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'NOT VersionNT64' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_SHAREDFX_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_SHAREDFX' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_FTS_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_FTS' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_X86_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_X86' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_LTS_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_LTS' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_RUNTIME_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_RUNTIME' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_SHARED_CONFIG_CHECK_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_SHARED_CONFIG_CHECK' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'OPT_NO_ANCM_Exists' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i000: Setting string variable 'OPT_NO_ANCM' to value '0'
[2E70:28A0][2023-09-15T13:40:57]i107: Detected forward compatible bundle: {e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}, type: Upgrade, scope: PerMachine, version: 6.0.22.23424, enabled: No
[2E70:28A0][2023-09-15T13:40:57]i102: Detected related bundle: {00e7db3a-80dc-40ac-b853-9b189e9bd477}, type: Upgrade, scope: PerMachine, version: 6.0.21.23364, operation: None
[2E70:28A0][2023-09-15T13:40:57]i102: Detected related bundle: {e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}, type: Upgrade, scope: PerMachine, version: 6.0.22.23424, operation: None
[2E70:28A0][2023-09-15T13:40:57]i103: Detected related package: {C0BD9C29-17A7-32B9-8242-17CFCAA1DA4F}, scope: PerMachine, version: 6.0.22.23424, language: 0 operation: Downgrade
[2E70:28A0][2023-09-15T13:40:57]i108: Detected compatible package: AspNetCoreModuleV2_x64, provider: IIS_AspNetCore_Module_V2,x64, installed: {97BB1873-8AF1-4B6B-9812-7BC4400C04B4}, version: 16.0.23237.0, chained: {31F32B18-FD5C-413A-91DA-E6ACBB99FB22}
[2E70:28A0][2023-09-15T13:40:57]i103: Detected related package: {97BB1873-8AF1-4B6B-9812-7BC4400C04B4}, scope: PerMachine, version: 16.0.23237.0, language: 0 operation: Downgrade
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'DotNetRedistLtsProductVersion_x64 = v48.67.58427' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'DotNetRedistLtsProductVersion_x86 = v48.67.58427' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]w120: Detected partially cached package: SharedFxRedist_x64, invalid payload: SharedFxRedist_x64, reason: 0x80070002
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'SharedFxRedistProductVersion_x64 = v6.0.16.23174' evaluates to true.
[2E70:28A0][2023-09-15T13:40:57]i052: Condition 'SharedFxRedistProductVersion_x86 = v6.0.16.23174' evaluates to false.
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: WindowsServerHostingBundleOptions.msi, state: Obsolete, cached: Complete
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: AspNetCoreModuleV2_x86, state: Absent, cached: None
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: AspNetCoreModuleV2_x64, state: Obsolete, cached: Complete
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: DotNetRedistLts_x64, state: Absent, cached: Complete
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: DotNetRedistLts_x86, state: Absent, cached: Complete
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: SharedFxRedist_x64, state: Present, cached: Partial
[2E70:28A0][2023-09-15T13:40:57]i101: Detected package: SharedFxRedist_x86, state: Absent, cached: Complete
[2E70:28A0][2023-09-15T13:40:57]i199: Detect complete, result: 0x0
[2E70:4110][2023-09-15T13:40:57]i000: Disable control ModifyNoIIS
[2E70:28A0][2023-09-15T13:40:59]i200: Plan begin, 7 packages, action: Uninstall
[2E70:28A0][2023-09-15T13:40:59]w209: Plan skipped removal of provider key: 76df85e0-b371-31e1-b3f1-a1c643d9acbd because it is registered to a different bundle: {e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}
[2E70:28A0][2023-09-15T13:40:59]w321: Skipping dependency registration on package with no dependency providers: SharedFxRedist_x86
[2E70:28A0][2023-09-15T13:40:59]w321: Skipping dependency registration on package with no dependency providers: SharedFxRedist_x64
[2E70:28A0][2023-09-15T13:40:59]i000: Setting string variable 'WixBundleLog_SharedFxRedist_x64' to value 'C:\Users\DFi\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20230915134057_000_SharedFxRedist_x64.log'
[2E70:28A0][2023-09-15T13:40:59]i000: Setting string variable 'WixBundleRollbackLog_SharedFxRedist_x64' to value 'C:\Users\DFi\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20230915134057_000_SharedFxRedist_x64_rollback.log'
[2E70:28A0][2023-09-15T13:40:59]w321: Skipping dependency registration on package with no dependency providers: DotNetRedistLts_x86
[2E70:28A0][2023-09-15T13:40:59]w321: Skipping dependency registration on package with no dependency providers: DotNetRedistLts_x64
[2E70:28A0][2023-09-15T13:40:59]w327: Will not uninstall package: {97BB1873-8AF1-4B6B-9812-7BC4400C04B4}, found dependents: 2
[2E70:28A0][2023-09-15T13:40:59]w328: Found dependent: {00e7db3a-80dc-40ac-b853-9b189e9bd477}, name: Unknown
[2E70:28A0][2023-09-15T13:40:59]w328: Found dependent: {e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}, name: Unknown
[2E70:28A0][2023-09-15T13:40:59]i207: Planned related bundle: {e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}, type: Upgrade, default requested: None, ba requested: None, execute: None, rollback: None, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i207: Planned related bundle: {00e7db3a-80dc-40ac-b853-9b189e9bd477}, type: Upgrade, default requested: None, ba requested: None, execute: None, rollback: None, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: SharedFxRedist_x86, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: Yes, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: SharedFxRedist_x64, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, cache: Yes, uncache: Yes, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: DotNetRedistLts_x86, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: Yes, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: DotNetRedistLts_x64, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: Yes, dependency: None
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: AspNetCoreModuleV2_x64, state: Obsolete, default requested: None, ba requested: None, execute: None, rollback: None, cache: No, uncache: Yes, dependency: Unregister
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: AspNetCoreModuleV2_x86, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: Unregister
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: WindowsServerHostingBundleOptions.msi, state: Obsolete, default requested: None, ba requested: None, execute: None, rollback: None, cache: No, uncache: Yes, dependency: Unregister
[2E70:28A0][2023-09-15T13:40:59]i201: Planned package: {97BB1873-8AF1-4B6B-9812-7BC4400C04B4}, state: Present, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: Unregister
[2E70:28A0][2023-09-15T13:40:59]i299: Plan complete, result: 0x0
[2E70:28A0][2023-09-15T13:40:59]i300: Apply begin
[2E70:28A0][2023-09-15T13:40:59]i010: Launching elevated engine process.
[2E70:28A0][2023-09-15T13:40:59]i011: Launched elevated engine process.
[2E70:28A0][2023-09-15T13:40:59]i012: Connected to elevated engine.
[1F0C:2414][2023-09-15T13:40:59]i358: Pausing automatic updates.
[1F0C:2414][2023-09-15T13:40:59]i359: Paused automatic updates.
[1F0C:2414][2023-09-15T13:40:59]i360: Creating a system restore point.
[1F0C:2414][2023-09-15T13:40:59]i361: Created a system restore point.
[1F0C:2414][2023-09-15T13:40:59]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}, resume: Active, restart initiated: No, disable resume: No
[2E70:12F8][2023-09-15T13:40:59]i336: Acquiring container: WixAttachedContainer, copy from: C:\ProgramData\Package Cache\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}\dotnet-hosting-6.0.16-win.exe
[2E70:1630][2023-09-15T13:40:59]e000: Error 0x80070001: Failed to extract all files from container, erf: 1:2:0
[2E70:12F8][2023-09-15T13:40:59]e000: Error 0x80070001: Failed to wait for operation complete.
[2E70:12F8][2023-09-15T13:40:59]e000: Error 0x80070001: Failed to open container.
[2E70:12F8][2023-09-15T13:40:59]e000: Error 0x80070001: Failed to open container: WixAttachedContainer.
[2E70:12F8][2023-09-15T13:40:59]e312: Failed to extract payloads from container: WixAttachedContainer to working path: C:\WINDOWS\Temp\{DC070D31-FE2C-4041-8DB3-58CA9B71A3C5}\80FBA19AB228FE43A71C5F2785B352309540C6ED, error: 0x80070001.
[2E70:28A0][2023-09-15T13:40:59]e000: Error 0x80070001: Cache thread exited unexpectedly.
[1F0C:2414][2023-09-15T13:40:59]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}, resume: ARP, restart: None, disable resume: No
[1F0C:2414][2023-09-15T13:40:59]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{58f3f479-fe1b-4cb4-b547-d6cb9196eac4}, resume: ARP, restart initiated: No, disable resume: No
[2E70:28A0][2023-09-15T13:41:00]i399: Apply complete, result: 0x80070001, restart: None, ba requested restart:  No
`

How can I uninstall all dotnet-hosting-bundle-versions?

### Reproduction Steps

Install dotnet-hosting-bundles versions
  6.0.22
  6.0.21
  6.0.16

run `dotnet-core-uninstall remove --hosting-bundle 6.0.22 --hosting-bundle 6.0.21 --hosting-bundle 6.0.16 `.

After failing try to run `.\dotnet-hosting-6.0.16-win.exe /uninstall` in the corresponding cache folder.

### Expected behavior

All hosting bundles get uninstalled

### Actual behavior

Nothing gets uninstalled.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
Did you try to uninstall it from the Add/ Remove programs list in Windows?

**Comment by WillBaldwin** [comment]
I have the very similar issue:

.NET Core Runtime & Hosting Bundles:
  6.0.5
  6.0.22
  6.0.21
  6.0.16
  6.0.12

Unable to uninstall 6.0.5, 6.0.12, 6.0.16, 6.0.21.

1) I tried "**using: dotnet-core-uninstall remove 6.0.5 --hosting-bundle**".  I received "The following items will be removed: Microsoft .NET 6.0.5 - Windows Server Hosting", this failed silently with no discernable errors.  
  
2) I also tried uninstalling using add remove programs, this also failed silently with no discernable errors.  
  
3) I re-attempted step 1) and step 2) with the IIS service shutdown just in case it was holding a DLL in memory, this also failed silently with no discernable errors.

**Comment by dfischerinsiderstechnologies** [comment]
I solved the issue: The "dotnet-core-uninstall"-tool tried to uninstall the hosting bundles using "dotnet-hosting-x.x.xx.exe"-files from a cache folder location e. g.: "C:\ProgramData\Package Cache\{e9c5b436-c2f7-4a9e-8a63-55c9e4eb1bcb}\dotnet-hosting-6.0.22-win.exe". These files were either corrupted or didn't work for whatever different reason. 

I downloaded my desired bundle again from here: https://dotnet.microsoft.com/en-us/download/dotnet/6.0 
I used those newly downloaded files to uninstall the bundles and it worked fine.
